File: artifacts/tulipindicators/translate_without_break_down/src/decay.rs
use crate::*;

pub fn ti_decay_start(options: Option<&[f64]>) -> i32 {
    let _ = options; // Explicitly ignore the parameter to match C behavior
    0
}
pub fn ti_decay(
    size: i32,
    inputs: &[&[f64]],
    options: &[f64],
    outputs: &mut [&mut [f64]],
) -> i32 {
    let input = inputs[0];
    let output = &mut outputs[0];
    let mut output_idx = 0;
    let period = options[0] as i32;
    let scale = 1.0 / period as f64;
    
    output[output_idx] = input[0];
    output_idx += 1;
    
    for i in 1..size as usize {
        let d = output[output_idx - 1] - scale;
        output[output_idx] = if input[i] > d { input[i] } else { d };
        output_idx += 1;
    }
    
    0
}

File: artifacts/tulipindicators/decompose/src/decay.c
int ti_decay_start(const double *options)
{
  (void) options;
  return 0;
}


/*** DEPENDENCIES:
***/


int ti_decay(int size, const double * const *inputs, const double *options, double * const *outputs)
{
  const double *input = inputs[0];
  double *output = outputs[0];
  unsigned int output_idx = 0;
  const int period = (int) options[0];
  const double scale = 1.0 / period;
  output[output_idx] = input[0];
  output_idx += 1;
  int i;
  for (i = 1; i < size; i += 1)
  {
    double d = output[(-1) + output_idx] - scale;
    output[output_idx] = (input[i] > d) ? (input[i]) : (d);
    output_idx += 1;
  }

  return 0;
}


/*** DEPENDENCIES:
***/


